happi/client.py

```diff
@@ -535,3 +535,34 @@ def find_config():
         # If found nothing
         raise EnvironmentError("No happi configuration file found. "
                                "Check HAPPI_CFG.")
+
+    def list_choices(self, field):
+        """
+        List all choices for a given field
+
+        Parameters
+        ----------
+        field : string
+            search field to list all possible choices for
+            i.e 'beamline', 'name', 'z', 'prefix', etc.
+
+        Raises
+        ------
+        SearchError
+            If no devices in the database have an entry for the given field
+
+        Returns
+        -------
+        field_choices : list
+            list of choices for a given field that are in the database
+        """
+        field_choices = []
```

Comment:
Python has a data structure for this sort of thing called `Set`. It is essentially an unordered `List` that only allows unique values. I think it could be of some use to you here. 

https://www.w3schools.com/python/python_sets.asp

Reply:
Changed in 2191d86